### PR TITLE
feat(key-auth): support get key from variable combination

### LINKED ISSUE #7530: feat: As a user, I want to authenticate clients by client_id and client_secret headers
### Description

We would like to authenticate clients by client_id and client_secret headers. But one key-auth plugin offers just one header to use. I tried to connect 2 key-auth plugins in plugin orchestration section but that didn't work

### PR DESCRIPTION
**Pull request by soulbird**
### Description

Fixes #7530 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/plugin/key-auth.t [hunk 1]
```diff
@@ -585,3 +585,84 @@ GET /hello?auth=auth-one
...
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 28: verify auth get key  from var_combination
```
Comment (tzssangglass) [inline t/plugin/key-auth.t:661]:
```suggestion
=== TEST 28: verify auth get key from var_combination
```

File: docs/en/latest/plugins/key-auth.md [hunk 2]
```diff
@@ -45,8 +45,9 @@ For Route:
...
-| header | string | optional    | apikey  |       | The header to get the key from.                                                                                                                                                                                                                                               |
-| query  | string | optional    | apikey  |       | The query string to get the key from. Lower priority than header.                                                                                                                                                                                                             |
+| var_combination | string | optional |      |    | Set which variable combination we get the key from. For example, "$remote_addr $consumer_name" will be combined into a key after parsing. highest priority.                  |
```
Comment (moonming) [inline docs/en/latest/plugins/key-auth.md:48]: no test cases for `priority`

File: t/plugin/key-auth.t [hunk 3]
```diff
@@ -585,3 +585,84 @@ GET /hello?auth=auth-one
...
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "plugins": {
+                        "key-auth": {
+                            "var_combination": "${http_client_id}:${http_client_secret}"
```
Comment (moonming) [inline t/plugin/key-auth.t:633]: The example in the document is "$remote_addr $consumer_name", and there is an extra colon in the test case. So which one is right?

File: docs/en/latest/plugins/key-auth.md [hunk 4]
```diff
@@ -45,8 +45,9 @@ For Route:
...
+| header | string | optional    | apikey  |       | The header to get the key from. Lower priority than `var_combination`.                                                                                                                                                                                                                                              |
```
Comment (moonming) [inline docs/en/latest/plugins/key-auth.md:49]: need to add example in doc.

**Review by moonming — CHANGES_REQUESTED** [review]

File: apisix/plugins/key-auth.lua [hunk 5]
```diff
@@ -90,12 +93,28 @@ end
...
+        local err, n_resolved
+        key, err, n_resolved = core.utils.resolve_var(conf.var_combination, ctx.var)
+        if err then
+            core.log.error("could not resolve vars in ",
+                                conf.var_combination, " error: ", err)
+        end
+
+        if n_resolved == 0 then
+            key = nil
+        end
```
Comment (moonming) [inline apisix/plugins/key-auth.lua:108]:
```suggestion
        if err then
            core.log.error("could not resolve vars in ",
                                conf.var_combination, " error: ", err)

        elseif n_resolved == 0 then
            key = nil
        end
```

File: t/plugin/key-auth.t (hunk 3 above)
Comment (soulbird) [reply t/plugin/key-auth.t:633]: The documentation is just an example of what can be configured, it is not the same as the test case, and the documentation and the test case are not the same for the users. There is no right or wrong here. But I can unify them.

Comment (moonming) [reply t/plugin/key-auth.t:633]: This confuses me, how should I configure it?

Comment (soulbird) [reply t/plugin/key-auth.t:633]: OK. I'll add some examples and test cases later.

**Comment by soulbird** [comment]
This function is not needed temporarily, reopen it when necessary.

**Pull request closed without merging**